### ISSUE TITLE
Authentication Problem with Android 11

### ISSUE THREAD

**Issue by AlexKashi** [aws-amplify/aws-sdk-android#2616]
When signing up on android 11 the following error is logged:

 `E/UserContextDataProvider: Exception in creating JSON from context data`
  
Tested on On Pixel 4 Android 11 simulator, works fine on older versions of android.

This error is blocking users from signing in in production. How can it be resolved?

```
AWSMobileClient.getInstance().confirmSignUp(username, "000000", object : Callback<SignUpResult?> {
            override fun onResult(result: SignUpResult?) {
                Timber.d("User exists! %s", result)
                ThreadUtils.runOnUiThread {
                    progressBar.visibility = View.GONE
                    disclaimer?.visibility = View.GONE
                }
                userExists.onResult(true)
            }

            override fun onError(e: Exception) {
                ThreadUtils.runOnUiThread {
                    progressBar.visibility = View.GONE
                    disclaimer?.visibility = View.GONE
                }
                if (e is UserNotFoundException) {
                    Timber.d("User Not found")
                    userExists.onResult(false)
                } else if (e is NotAuthorizedException) {
                    Timber.d("User Not auth: %s", e.message)
                    userExists.onResult(true)
                } else if (e is AliasExistsException) {
                    Timber.d("AliasExistsException: %s", e.message)
                    userExists.onResult(true)
                } else if (e is CodeMismatchException) {
                    Timber.d("CodeMismatchException: %s", e.message)
                    userExists.onResult(true)
                } else if (e is ExpiredCodeException) {
                    Timber.d("ExpiredCodeException: %s", e.message)
                    userExists.onResult(true)
                } else if (e is LimitExceededException) {
                    ThreadUtils.runOnUiThread { Toast.makeText(context, "Limit exceeded, try again later.", Toast.LENGTH_LONG).show() }
                    userExists.onResult(null)
                } else {
                    Timber.e("Unknown Exception: %s", e.message)
                    userExists.onResult(null)
                }
            }
        })
````

My AWS version is 2.30.0 implementing:
"com.amazonaws:aws-android-sdk-cognitoidentityprovider:2.30.0"
"com.amazonaws:aws-android-sdk-cognitoauth:2.30.0"

related to aws-amplify/amplify-android#948

**Comment by noarazvia** [comment]
@rjuliano is there any ETA for this to be solved?

**Comment by mpanella** [comment]
some solution to this problem?

**Comment by dcap0** [comment]
@rjuliano Any update on this issue?

**Comment by eeatonaws** [comment]
Thanks for reporting this. We are working on determining the cause of the issue and will keep you updated here.

**Comment by taouichaimaa** [comment]
having the same problem in SignUp, emulating on android 11 and aws version 2.35.0. keeps users from creating their accounts. Any update?

**Comment by sandfox** [comment]
Adding some info based on some internal debugging. 

- The bug very much looks like it is originating from the  `aws-android-sdk-cognitoidentityprovider-asf` jar which is called if your cognito pool has advanced security enabled and/of if you have that setting enabled in your mobile app.
- The `aws-android-sdk-cognitoidentityprovider-asf` jar internally calls various android api's to generate a fingerprint fot the device, some of  which are documented as requirering `READ_PHONE_STATE` to be accessed. (see here: https://github.com/sandfox/aws-android-sdk-cognitoidentityprovider-asf/blob/master/com/amazonaws/cognito/clientcontext/datacollection/TelephonyDataCollector.java)
- It seems that if you build your android app using `targetSdkVersion` <= `29` then you do not infact need the `READ_PHONE_STATE` permission to access those android apis, no matter what version of android the app is running on. If you build your app using `targetSdkVersion` = `30` , then the app will require the `READ_PHONE_STATE` permission to access those api's when running on an android 11 devices.

Hopefully this will explain the behaviour some of you are seeing.

**Comment by eeatonaws** [comment]
A PR with the fix for this issue has been released as part of AWS SDK for Android version 2.38.0.
